alternator/executor.cc

```diff
@@ -2817,22 +2827,31 @@ future<executor::request_return_type> executor::batch_write_item(client_state& c
                 co_return api_error::validation(fmt::format("Unknown BatchWriteItem request type: {}", r_name));
             }
         }
+        if (should_add_wcu) {
+            rjson::value entry = rjson::empty_object();
+            rjson::add(entry, "TableName", rjson::from_string(rjson::to_string_view(it->name)));
+            rjson::add(entry, "CapacityUnits", wcu_units);
+            rjson::push_back(consumed_capacity, std::move(entry));
+        }
     }
-
+    _stats.wcu_total[stats::PUT_ITEM] += wcu_put_units;
+    _stats.wcu_total[stats::DELETE_ITEM] += wcu_delete_units;
```

Comment:
Maybe move this a couple more lines down, to be after the permission checks? Because if those fail we should probably not charge for the request as well.

See below where we increment `_stats.api_operation.batch_write_item_batch_total`. I think that's where should increment the wcu too.

Reply:
ok